sycl/source/detail/persistent_device_code_cache.cpp

```diff
@@ -339,13 +341,15 @@ std::string PersistentDeviceCodeCache::getCacheItemPath(
   if (Img.getRawData().BinaryStart)
     ImgString.assign((const char *)Img.getRawData().BinaryStart, Img.getSize());
 
-  std::string DeviceString{getDeviceIDString(Device)};
-  std::string SpecConstsString{(const char *)SpecConsts.data(),
-                               SpecConsts.size()};
-  std::hash<std::string> StringHasher{};
+  std::string_view ImgStringView(ImgString);
```

Comment:
No need to create these string views here, we can keep using `std::hash<std::string>` for those while switching to `std::hash<std::string_view>` for build options.

Reply:
We can certainly do that. I was just wondering if creating these string_views here and changing the hash to take string_view has any negative impact?

Reply:
No negative impact aside from creating unneeded string views when we can make do with already available strings.